src/CalculationClasses/MultiLayer/ReflectionMultiLayer.m

```diff
@@ -36,35 +35,34 @@ function Calculate(obj)
 
             obj.Setup_energy_mesh_full;
             obj.RecalculateToEnergyDistribution;
-            if obj.vacuum
-                R = zeros(size(obj.FullEnergyDistribution{obj.N_Layer}.R(:,:,:,:)));
-            else
-                R = obj.FullEnergyDistribution{obj.N_Layer}.R(:,:,:,:);
-            end
+            R = obj.FullEnergyDistribution{obj.N_Layer}.R(:,:,:,:); %bulk (semiinf) signal
             M = size(R,4)-1;
             N_E =size(R,3);
             wT = zeros(size(obj.FullEnergyDistribution{1}.T));
             Tw = zeros(size(obj.FullEnergyDistribution{1}.T));
             for i_layer = (obj.N_Layer-1):-1:1
-                Rlocal = obj.FullEnergyDistribution{i_layer}.R(:,:,:,:);
+                if obj.vacuum && i_layer == 1

```

Comment:
what if w have several vacuum layers with different inelastic properties?

Reply:
This is questionable, it is not the case for IEM, I'm not sure that we can use it in the same way like in MC for example.... I suppose that we can have only one layer with an "average" DIIMFP there. This will be eliminated within this PR.